integration-tests/jms/src/main/java/io/quarkus/it/artemis/ArtemisProducerManager.java

```diff
@@ -5,17 +5,51 @@
 import javax.jms.ConnectionFactory;
 import javax.jms.JMSContext;
 import javax.jms.JMSProducer;
+import javax.jms.XAConnectionFactory;
+import javax.jms.XAJMSContext;
+import javax.transaction.Synchronization;
+import javax.transaction.TransactionManager;
 
 @ApplicationScoped
 public class ArtemisProducerManager {
 
     @Inject
     ConnectionFactory connectionFactory;
 
+    @Inject
+    XAConnectionFactory xaConnectionFactory;
+
+    @Inject
+    TransactionManager tm;
+
+    private void _send(JMSContext context, String body) {
+        JMSProducer producer = context.createProducer();
+        producer.send(context.createQueue("test-jms"), body);
+    }
+
     public void send(String body) {
         try (JMSContext context = connectionFactory.createContext(JMSContext.AUTO_ACKNOWLEDGE)) {
-            JMSProducer producer = context.createProducer();
-            producer.send(context.createQueue("test-jms"), body);
+            _send(context, body);
+        }
+    }
+
+    public void sendXA(String body) throws Exception {
+        XAJMSContext context = xaConnectionFactory.createXAContext();
```

Comment:
shouldn't this also be in a try with resources?

Reply:
Anyway, this should be better processed by other wrapper, such as `org.messaginghub:pooled-jms`.

Reply:
Okay, not an expert on XA and it is test code, so should be fine.